feat: 🚸 adds confirmation dialog before user sends error report

jalezi opened the pull request:
Adds confirmation dialog before a user sends an error report.

![Screenshot 2024-02-26 at 09 20 43](https://github.com/sledilnik/zdravniki/assets/44704999/5c42bfd4-b8ea-4613-93e8-54a128b40658)

@SterArcher please check translations.

FIX: #447

stefanb commented on src/locales/sl.json:
```diff
@@ -79,6 +79,11 @@
+        },
+        "confirmation": {
+            "title": "Potrditev",
+            "warningText": "Sporočanje napak ni namenjeno naročanju na pregled ali posvet. Za naročanje se obrnite na izbranega zdravnika.",
+            "text": "Ali ste prepričani, da želite poslati sporočilo?"
```
```suggestion
            "warningText": "Sporočanje napak NI namenjeno naročanju na pregled ali posvet. Za naročanje se obrnite na izbranega zdravnika.",
            "text": "Ali ste prepričani, da želite predlagati popravek podatkov?"
```

stefanb commented on src/locales/en.json:
```diff
@@ -77,6 +77,11 @@
+        },
+        "confirmation": {
+            "title": "Confirm",
+            "warningText": "Error reporting is not intended to order a review or consultation. To order appointment, contact your chosen doctor.",
+            "text": "Are you sure you want to report an error?"
```
```suggestion
            "warningText": "This form is NOT intended to request an appointment for medical examination or consultation. To make an appointment, contact your chosen doctor.",
            "text": "Are you sure you want to suggest a data update?"
```

stefanb commented on src/locales/it.json:
```diff
@@ -73,7 +73,12 @@
+        "send": "Invia",
+        "confirmation": {
+            "title": "Confirma l'invio",
+            "text": "Sei sicuro di voler inviare la segnalazione?",
+            "warningText": "La segnalazione degli errori non ha lo scopo di ordinare una revisione o una consultazione. Per ordinare, contattare il medico prescelto."
```
This probably requires a similar update as English and Slovenian. It may be better to leave it untranslated, to show it in English until it is translated by translators.
```suggestion
```

stefanb commented on src/locales/it.json:
```diff
@@ -73,7 +73,12 @@
         "title": "Segnala un errore",
         "cancel": "Annulla",
-        "send": "Invia"
+        "send": "Invia",
+        "confirmation": {
+            "title": "Confirma l'invio",
+            "warningText": "Error reporting is not intended to order a review or consultation. To order appointment, contact your chosen doctor.",
+            "text": "Are you sure you want to report an error?"
+        }
```
The translations should not be present at all:
```suggestion
        "send": "Invia"
```
so that they are marked as "untranslated" in weblate, but until then the default (English) texts will be shown.

Also you can use this suggestion by clicking "Commit suggestion" button bellow ⬇️ this comment 😄

stefanb commented on src/components/DoctorCard/ReportError/index.js:
```diff
@@ -192,44 +217,56 @@ const ReportError = function ReportError({ doctorFormData, setIsEditing, setMess
           />
-          <SelectEdit name="inputAccepts" value={inputAccepts} setValue={setInputAccepts} />
+          <SelectEdit
+            id="inputAddress"
```
Copy-paste error?
```suggestion
            id="inputAccepts"
```

stefanb commented:
Tested by confirming 2 doctor changes and all the fields except timestamp came empty to the gsheet, including the basic data and the note that i wrote:
<img width="704" alt="image" src="https://github.com/sledilnik/zdravniki/assets/319826/c95399fc-66b5-488d-be36-3f07d8ce192d">

stefanb reviewed (changes requested):
no data comes to gsheet

jalezi commented:
> Tested by confirming 2 doctor changes and all the fields except timestamp came empty to the gsheet, including the basic data and the note that i wrote: <img alt="image" width="704" src="https://private-user-images.githubusercontent.com/319826/308663541-c95399fc-66b5-488d-be36-3f07d8ce192d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDU3MTMsIm5iZiI6MTcwOTE0NTQxMywicGF0aCI6Ii8zMTk4MjYvMzA4NjYzNTQxLWM5NTM5OWZjLTY2YjUtNDg4ZC1iZTM2LTNmMDdkOGNlMTkyZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyOFQxODM2NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGU2ZGMyNGQxOTZjYzczMjhiZmVhM2VjNjI5MzgxNjZjYzAzYzVjMmY4MzI4MmQzNWVkZjU5MmRlNDEzNzRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.el4FHoFHgpODPCMI9pKfabfRhzyVxe4alR9iKbfN8eQ">

@stefanb give link me to gsheet. It should work. env vars?

```json
{
    "inputName": {
        "id": "1889437249",
        "value": "Ahačič Srednik Nika"
    },
    "inputUrl": {
        "id": "1318563412",
        "value": "http://localhost:3000/sl/gp/ahacic-srednik-nika/141819"
    },
    "inputType": {
        "id": "2146648493",
        "value": "gp"
    },
    "inputInstId": {
        "id": "",
        "value": "141819"
    },
    "inputProvider": {
        "id": "480165709",
        "value": "ZDRAVSTVENI DOM OSNOVNO VARSTVO NOVA GORICA"
    },
    "inputFullAddress": {
        "id": "1560677043",
        "value": "Rejčeva ulica 4, 5000 Nova Gorica"
    },
    "inputAccepts": {
        "id": "1569335568",
        "value": "n"
    },
    "inputAvailability": {
        "id": "77408979",
        "value": "100 %"
    },
    "inputWebsite": {
        "id": "1504147301",
        "value": "http://www.zd-go.si"
    },
    "inputPhone": {
        "id": "1724437941",
        "value": "041345678"
    },
    "inputEmail": {
        "id": "",
        "value": "example@example.com"
    },
    "inputOrderform": {
        "id": "",
        "value": "example.com"
    },
    "inputNote": {
        "id": "1714314704",
        "value": "test - jaka"
    }
}
```

stefanb commented:
I was looking at production gsheet, but it is interesting, that the timestamp made it there, but nothing else.

jalezi commented:
> I was looking at production gsheet, but it is interesting, that the timestamp made it there, but nothing else

My bad, I made a rookie mistake. 😥

SterArcher commented:
I have checked the translation I would change it to say:
“Sporocanje napak ni namenjeno narocanju na pregled ali opredelitev pri zdravniku. Za narocanje se obrnite na izbranega zdravnika preko prikazanih komunikacijskih kanalov.”

This all looks great, thanks!

jalezi commented:
Sorry, @stefanb and @SterArcher I recognized very late that posting a form is working. 😁 

I refactored quite a lot. Added some functionality:
- The address field is required;
- A little bit better message alerts (error or no field has changed);
- The user can now reset the form without closing the form.
- The inputs have labels (only for screen readers)

Some translations are needed.

stefanb reviewed (approved):
LGTM!
